Change version of MathJax to 2.7

sethferrell opened the pull request:
Changing the version will provide more options for us as the newest version of MathJax has not updated all its components yet.
Doing so will not change any prior work we have done.

Mahdi03 committed:
Clarified on why (l) and (s) are omitted from equilibrium math

sethferrell committed:
Changed version of MathJax

Mahdi03 commented on renderMath.js:
```diff
@@ -223,7 +231,7 @@ function renderMath(parentElement = "") {
                     oldBR.replaceWith(...newBR);
                 }
-                oldSpanAlignedEquations.innerHTML = oldSpanAlignedEquations.innerHTML.replace(/=/g, "&=").replace(/\-\>/gm, "&->").replace(/\-&gt;/gm, "&->");
+                oldSpanAlignedEquations.innerHTML = oldSpanAlignedEquations.innerHTML.replace(/=/g, "&=");
```
Undo this change, it helps align the chemical equations
```suggestion
                oldSpanAlignedEquations.innerHTML = oldSpanAlignedEquations.innerHTML.replace(/=/g, "&=").replace(/\-\>/gm, "&->").replace(/\-&gt;/gm, "&->");
```

Mahdi03 commented on renderMath.js:
```diff
@@ -223,7 +231,7 @@ function renderMath(parentElement = "") {
                 }
-                oldSpanAlignedEquations.innerHTML = oldSpanAlignedEquations.innerHTML.replace(/=/g, "&=").replace(/\-\>/gm, "&->").replace(/\-&gt;/gm, "&->");
+                oldSpanAlignedEquations.innerHTML = oldSpanAlignedEquations.innerHTML.replace(/=/g, "&=");
                 var newSpanAlignedEquations = createHTMLNodesFromString("\\begin{align}" + oldSpanAlignedEquations.innerHTML + "\\end{align}");
```
On your end, change the `align` to `aligned` inside the MathJax strings, let me know if it renders correctly now
```suggestion
                var newSpanAlignedEquations = createHTMLNodesFromString("\\begin{aligned}" + oldSpanAlignedEquations.innerHTML + "\\end{aligned}");
```

However if that does not work, then try:

```suggestion
                var newSpanAlignedEquations = createHTMLNodesFromString("\\eqalign{" + oldSpanAlignedEquations.innerHTML + "}");
```

And if that doesn't work then give up with 2.7 and we're sticking with 3.x because who really needs to copy formulas from our study guide.

Mahdi03 commented on renderMath.js:
```diff
@@ -78,13 +87,12 @@ function renderMath(parentElement = "") {
+            // polyfillScript.setAttribute("src", "https://polyfill.io/v3/polyfill.min.js?features=es6");
             var mathjaxScript = document.createElement("script");
-            mathjaxScript.setAttribute("id", "MathJax-script");
+            mathjaxScript.setAttribute("type", "text/javascript");
```
Remove this line, what are we...HTML4?

Mahdi03 commented on renderMath.js:
```diff
@@ -78,13 +87,12 @@ function renderMath(parentElement = "") {
+            mathjaxScript.setAttribute("type", "text/javascript");
             mathjaxScript.setAttribute("async", "");
-            mathjaxScript.setAttribute("src", "https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js");
-            document.head.appendChild(polyfillScript);
```
If there is no polyfill script, either comment this line out like you did with the others, or delete the other two commented lines out above (regarding the polyfill script), we've gotta be consistent in the changes we make

Mahdi03 commented on renderMath.js:
```diff
@@ -78,13 +87,12 @@ function renderMath(parentElement = "") {
             mathjaxScript.setAttribute("async", "");
-            mathjaxScript.setAttribute("src", "https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js");
-            document.head.appendChild(polyfillScript);
+            mathjaxScript.setAttribute("src", "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML");
```
Is it possible to remove the attribute from the link? Adding a `?` and parameters to the link prevents caching, and caching speeds up the loading process. Our page is already really slow so having to download the mathjax script on every page load wouldn't be a help to that cause.
```suggestion
            mathjaxScript.setAttribute("src", "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js");
```

Mahdi03 commented on AP Chem/Unit 7.html:
```diff
@@ -36,7 +36,7 @@ <h5>The Equilibrium Constant:</h5>
         <li>Reversible reactions at equilibrium with different states of matter are said to be in <u>Heterogeneous Equilibrium</u>. In these cases we ignore the solvents and the solids because they are in excess, so it makes sense to not take into account
             such high concentrations into our formula for the equilibrium constant.</li>
-        <li>In equilibrium reactions, pay attention to the state of matter. <strong style="color:mediumvioletred;">Concentrations of solids and liquids are omitted from all calculations</strong>.</li>
+        <li>In equilibrium reactions, pay attention to the state of matter. <strong style="color:mediumvioletred;">Concentrations of solids and liquids are omitted from all calculations</strong>. Another way to think of this is that liquids and solids don't have concentrations to begin with, there is no such thing as stronger water.</li>
```
Oh hehe dude I added this line in class when Ms. Cooper was going over it&mdash;just thought you should know

sethferrell committed:
Update renderMath.js

Co-authored-by: Mahdi03 <damahdi03@gmail.com>

sethferrell committed:
Update renderMath.js

Co-authored-by: Mahdi03 <damahdi03@gmail.com>

sethferrell committed:
Update renderMath.js

Mahdi03 committed:
Merge branch 'beta' into sethtest

Mahdi03 committed:
Uhm no Seth (read on to find out more!)

Ok so I don't have the time or effort to actually make this work, I
have already been stuck on this for a while now, lemme explain what
the issue is:

- First of all, it says that I have 9 commits here, that's actually
  untrue, I just merged the `beta` branch into this separate branch
  so that I could have the most updated changes, but the changes that
  I did make are reflected in this commit
- Your MathJax 2.7 is loading correctly but it is not TYPESETTING any
  of the math which is the most important part, and MathJax 3.x WAS
  typesetting all the math (to typeset means to render the math all
  fancy-schmancy like how you wanted it)
- After spending two hours on this, I have found no way to actually
  take what we have and reinvent the wheel so that anything renders,
  I can't find a way to typeset the math so I don't know what kind of
  voodoo magic you worked to get it to run on your end, but it ain't
  working here, whereas the 3.x does
- If you can find the mistake then by all means fix it, I'll keep the
  branch alive but I am closing the pull request, I do not know how you
  got 2.7 to work on your machine but it is not working on mine (even
  with the Python Server)
- I have decided that we should not fix what is not broken, and so what
  if our users can't copy and paste our formulas, 3.x works and I've
  already put enough effort into rewriting my math thing for that
- However I will be taking the configuration changes into account in
  the `beta` branch

Anyways sorry for "flaming" or whatever that was, I'm just tired, let's
add some images/diagrams to this bad boy and then publish to the public

Mahdi03 commented:
Read the last commit
